pkg/sql/stats/histogram.go

```diff
@@ -170,8 +170,16 @@ type histogram struct {
 }
 
 // adjustCounts adjusts the row count and number of distinct values per bucket
-// based on the total row count and estimated distinct count.
+// to equal the total row count and estimated distinct count. The total row
+// count and estimated distinct count should not include NULL values, and the
+// histogram should not contain any buckets for NULL values.
 func (h *histogram) adjustCounts(evalCtx *eval.Context, rowCountTotal, distinctCountTotal float64) {
+	// Empty table cases.
+	if rowCountTotal <= 0 || distinctCountTotal <= 0 {
```

Comment:
I'm guessing negative counts are possible with forecasting too? Maybe the commit message should mention that. And you could add test cases with negative counts if you'd like.

Reply:
Good point, I added a test case.

The current PR [normalizes negative counts to zero](https://github.com/cockroachdb/cockroach/blob/c903654bbbb2d351625e28a5864766b5a83f8af6/pkg/sql/stats/forecast.go#L261) before calling `adjustCounts` but you're absolutely right, this was my intention with <=.